pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -287,6 +287,16 @@ public static String getStringValue(Expression thriftExpression) {
    *          missing an EQUALS filter operator.
    */
   public static FilterContext getFilter(ExpressionContext filterExpression) {
+    FunctionContext function = filterExpression.getFunction();
+    // Trim off outer IS_TRUE function as it is redundant
+    if (function != null && function.getFunctionName().equals("istrue")) {
+      return getFilter(function.getArguments().get(0));
+    } else {
```

Comment:
Shouldn't we check if the argument is also a function? Do we want to apply this optimization even if the argument is an identifier?

Reply:
`function != null` is equivalent to that check